core/server/admin/app.js

```diff
@@ -41,6 +41,9 @@ module.exports = function setupAdminApp() {
         {maxAge: utils.ONE_YEAR_MS, fallthrough: false}
     ));
 
+    // Service Worker for offline support
+    adminApp.get(/^\/(sw.js|sw-registration.js)$/, require('./serviceworker'));
```

Comment:
@kevinansfield is totally right - we need to serve it from that scope to allow the service worker to cache and intercept requests on that origin. If you're wondering why I'm not just doing a 301 redirect - even those aren't allowed.

This is mostly a security feature, ensuring that service workers only ever intercept (and potentially change) traffic on origins they have access to.